integration-tests/src/tests/bitbake-parse.test.ts

```diff
@@ -16,14 +16,30 @@ async function delay (ms: number): Promise<void> {
 
 suite('Bitbake Commands Test Suite', () => {
   let disposables: vscode.Disposable[] = []
+  let workspaceURI: vscode.Uri
+  let buildFolder: vscode.Uri
 
   suiteSetup(async function (this: Mocha.Context) {
-    this.timeout(10000)
+    this.timeout(100000)
     while (vscode.workspace.workspaceFolders === undefined || vscode.workspace.workspaceFolders?.length === 0) {
       await delay(100)
     }
+    workspaceURI = vscode.workspace.workspaceFolders[0].uri
+    buildFolder = vscode.Uri.joinPath(workspaceURI, 'build')
+
     // Generate the build directory for the addLayer functions to work
+    let taskExecuted = false
+    const disposable = vscode.tasks.onDidEndTask(async (e) => {
+      if (e.execution.task.definition.options.parseOnly === true) {
+        taskExecuted = true
+      }
+    })
     await vscode.commands.executeCommand('bitbake.parse-recipes')
```

Comment:
Actually this is not a Promise. executeCommand starts a Task which starts a Terminal and the promise only points to the started task, we can't get the outcome that way. That's why I'm using the onDidEndTask callback to detect task termination. I don't have access to wether it failed or not, and other tasks may be spawned because the bitbakeProjectScanner is running in the background. I don't think I can make this code look better with `then()`.
Currently, it has a nice sequential order of instructions despite the callbacks when chaining multiple commands.

However suggestions to make this simpler would be welcome.